pkg/platform/platform.go

```diff
@@ -805,3 +817,52 @@ func (platform *Platform) DefaultNamespaceAnnotations() map[string]string {
 func (platform *Platform) IsMaster(machine types.TagInterface) bool {
 	return machine.GetTags()["Role"] == platform.Name+"-masters"
 }
+
+// validateCluster is to validate common platform configs
+func validateCluster(platform *Platform) error {
+	if platform.CA == nil {
+		return fmt.Errorf(ErrNoCaSection)
+	}
+	if platform.IngressCA == nil {
+		return fmt.Errorf(ErrNoIngressCaSection)
+	}
+	if _, err := platform.GetCA(); err != nil {
+		return errors.Wrap(err, ErrInvalidCA)
+	}
+
+	return nil
+}
+
+// ValidateKindCluster is to validate a platform config
+func (platform *Platform) ValidateKindCluster() error {
+	err := validateCluster(platform)
+	if err != nil {
+		return err
+	}
+	if platform.Kubernetes.Version == "" {
+		return fmt.Errorf(ErrK8sVersionUnset)
+	}
+	return nil
+}
+
+// ValidateVSphereCluster is to validate a platform config
+func (platform *Platform) ValidateVSphereCluster() error {
+	err := validateCluster(platform)
+	if err != nil {
+		return err
+	}
+
+	if platform.Master.CPUs < 2 {
+		return fmt.Errorf(ErrInsufficientVms)
+	}
+
+	if platform.Calico.Version == "" {
```

Comment:
Either a calico or nsx version needs to be specified

Reply:
see if https://github.com/flanksource/karina/pull/365/commits/7702f5754edb6089ccf05e1516ec9bfdc5851eb4 resolves this satisfactorily